### PR TITLE
fix(sdk-coin-eth): fix v1 address init code

### PR DESCRIPTION
**Pull request by gianchandania**
This PR modifies logic to initialize v1 addresses

**Changes**

- Modifies `buildAddressInitializationTransaction()` method to support v1 address initialization

- Modifies `setTransactionTypeFields() `method to support decoding of v1 addresses

- Added `decodeCreateV1ForwarderData()` method to support decoding of v1 addresses

- Added `getV1AddressInitializationData()` method to get method call data for v1 addresses initialization

- Added `baseAddress()` method to set the baseAddress of wallet

- Modifies `getAddressInitializationData()` by removing optional param forwarderVersion. This method now only returns the method id for createForwarder method for v0 forwarder and not v1 forwarders

- Added tests for addressInitialization for polygon

- Modifies tests because of  change in method `getAddressInitializationData()`

- Renamed `getWalletV1InitData()` added in this [PR](https://github.com/BitGo/BitGoJS/pull/2607) for consistency 

- Renamed createWalletTypes added in this [PR](https://github.com/BitGo/BitGoJS/pull/2607) for consistency

Ticket: [BG-53920](https://bitgoinc.atlassian.net/browse/BG-53920)

### CODE REVIEW

File: modules/sdk-coin-eth/src/lib/utils.ts [hunk 1]
```diff
@@ -190,12 +191,10 @@ export function flushCoinsData(): string {
...
  * Returns the create forwarder method calling data
  *
- * @param {string} forwarderVersion - forwarder version for the address to be initialized
  * @returns {string} - the createForwarder method encoded
  */
-export function getAddressInitializationData(forwarderVersion?: number): string {
-  const impliedForwarderVersion = forwarderVersion ?? defaultForwarderVersion;
-  return impliedForwarderVersion === defaultForwarderVersion ? createForwarderMethodId : v1CreateForwarderMethodId;
+export function getAddressInitializationData(): string {
+  return createForwarderMethodId;
```
Comment (gianchandania) [inline modules/sdk-coin-eth/src/lib/utils.ts:197]: @mmcshinsky-bitgo I am wondering if removing an optional param is a breaking change?

Comment (mmcshinsky-bitgo) [reply modules/sdk-coin-eth/src/lib/utils.ts:205]: Only required parameters are breaking changes, but we should clearly detail the change in the commit description.

Comment (gianchandania) [reply modules/sdk-coin-eth/src/lib/utils.ts:205]: Got it. Will update the description

File: modules/sdk-coin-eth/src/lib/utils.ts [hunk 2]
```diff
@@ -61,8 +64,12 @@ import {
   walletInitializationFirstBytes,
   v1CreateForwarderMethodId,
   walletSimpleConstructor,
+<<<<<<< HEAD
   createWalletTypes,
   v1CreateWalletMethodId,
+=======
+  createV1ForwarderTypes,
+>>>>>>> 27bae27d3 (fix(sdk-coin-eth): fix v1 address init code)
```
Comment (mmcshinsky-bitgo) [inline modules/sdk-coin-eth/src/lib/utils.ts:72]: Merge conflict comments need to be removed here.

**Review by mmcshinsky-bitgo — CHANGES_REQUESTED** [review]

File: modules/sdk-coin-eth/src/lib/utils.ts (hunk 2 above)
Comment (gianchandania) [reply modules/sdk-coin-eth/src/lib/utils.ts:72]: working on it

File: modules/sdk-coin-eth/src/lib/utils.ts (hunk 1 above)
Comment (mmcshinsky-bitgo) [reply modules/sdk-coin-eth/src/lib/utils.ts:205]: Does removing the optional parameter create situations where the end consumers will need to change their own code to get previous expectations that the function provided in the expected logic flow?

File: modules/sdk-coin-eth/src/lib/utils.ts (hunk 2 above)
Comment (gianchandania) [reply modules/sdk-coin-eth/src/lib/utils.ts]: done

File: modules/sdk-coin-eth/src/lib/utils.ts (hunk 1 above)
Comment (gianchandania) [reply modules/sdk-coin-eth/src/lib/utils.ts:199]: Actually to give a little background on the change, I had added this optional parameter in this [PR](https://github.com/BitGo/BitGoJS/pull/2515). But now I have made the function just like it was before this mentioned PR by removing the optional parameter. 

The method will behave in the same manner for the users who were not passing the optional parameter or were passing the default value of optional parameter. But if a non-default value was passed to the method, then it returned a different response previously, and now there are no 2 different responses. It is just a single response (i.e. for the default value).

**Review by mmcshinsky-bitgo — APPROVED** [review]

**Pull request merged**